ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -276,11 +298,24 @@ public boolean updateAccounts(Collection<Account> accounts) {
   @Override
   public void close() {
     if (isOpen) {
-      try {
-        isOpen = false;
+      isOpen = false;
+      if (scheduler != null) {
+        scheduler.shutdown();
+        try {
+          if (!scheduler.awaitTermination(storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS)) {
+            logger.warn("Scheduler for account updater is not shut down successfully");
+            scheduler.shutdownNow();
+            if (!scheduler.awaitTermination(storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS)) {
+              logger.error("Scheduler for account updater is not shut down successfully");
+            }
+          }
+        } catch (InterruptedException e) {
+          scheduler.shutdownNow();
+          Thread.currentThread().interrupted();
```

Comment:
log in this case

Reply:
Also, this should be `Thread.currentThread().interrupt()` not interrupted

Reply:
Added the log and fixed.